drainer/safepoint.go

```diff
@@ -0,0 +1,48 @@
+package drainer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb-binlog/drainer/checkpoint"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+const (
+	drainerServiceSafePointPrefix = "drainer"
+	defaultDrainerGCSafePointTTL  = 5 * 60
+)
+
+func updateServiceSafePoint(ctx context.Context, pdClient pd.Client, cpt checkpoint.CheckPoint, ttl int64) {
+	updateInterval := time.Duration(ttl/2) * time.Second
+	tick := time.NewTicker(updateInterval)
```

Comment:
tick.Stop

Reply:
addressed in https://github.com/pingcap/tidb-binlog/pull/1233/commits/fcd27dc32973a449dd6b94e0bb136c7a7fd6d9a3